### PR TITLE
Parallelize reference operation in TestSecretCacheMultipleRegistrations

### PR DESCRIPTION
**Pull request by tedyu**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
In TestSecretCacheMultipleRegistrations, currently AddReference / DeleteReference are done in serial.
However, since these operations are protected by lock, we should run them in parallel.

This PR uses go routines to do this.

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by tedyu** [comment]
cc @mattjmcnaughton

File: pkg/kubelet/util/manager/watch_based_manager_test.go [hunk 1]
```diff
@@ -166,10 +167,17 @@ func TestSecretCacheMultipleRegistrations(t *testing.T) {
...
-	for i := 0; i < 20; i++ {
-		store.AddReference("ns", "name")
-		store.DeleteReference("ns", "name")
-	}
+	wg := sync.WaitGroup{}
+	iterations := 20
+	wg.Add(iterations)
+	for i := 0; i < iterations; i++ {
+		go func() {
+			store.AddReference("ns", "name")
```
[the remark this answers is not in the document]
Comment (tedyu) [reply pkg/kubelet/util/manager/watch_based_manager_test.go]: @mattjmcnaughton 
Can you take another look ?

[the remark this answers is not in the document]
Comment (mattjmcnaughton) [reply pkg/kubelet/util/manager/watch_based_manager_test.go]: @tedyu can you share more about why the test would function correctly in the above case? I'm not sure I'm following.

**Pull request closed without merging**